packages/react/src/primitives/Checkbox/useCheckbox.ts

```diff
@@ -0,0 +1,28 @@
+import React, { ChangeEvent, useState } from 'react';
+
+export const useCheckbox = (
+  initialChecked: boolean,
+  onChangeProp: React.ChangeEventHandler
+) => {
+  const [dataChecked, setDataChecked] = useState(initialChecked);
+  const [dataFocus, setDataFocus] = useState(false);
+
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setDataChecked(e.target.checked);
+
+    // this is the onChange function user passing in
+    if (onChangeProp) {
```

Comment:
Check for function type first:
```suggestion
    if (isFn(onChangeProp)) {
```

Reply:
I am assuming you have added this function after I update with main